### PR TITLE
More fixes for crowbar_register on sle12

### PR DESCRIPTION
**Pull request by MaximilianMeister**

### CODE REVIEW

**Comment by vuntz** [comment]
+1

File: chef/cookbooks/provisioner/templates/suse/crowbar_register.erb [hunk 1]
```diff
@@ -193,8 +193,13 @@ zypper --non-interactive install $PACKAGES_INSTALL
 /sbin/SuSEfirewall2 off
 
 # SSH setup
+<% if @target_platform_version.to_f < 12.0 %>
```
Comment (vuntz) [inline chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: Actually, can you change `%>` to `-%>`?

File: chef/cookbooks/provisioner/templates/suse/crowbar_register.erb [hunk 2]
```diff
@@ -311,7 +316,11 @@ sed -i -e "s/\(127\.0\.0\.1.*\)/127.0.0.1 localhost.localdomain localhost/" /etc
 echo "$CROWBAR_KEY" > /etc/crowbar.install.key
 
 # normally done by crowbar_join, but only when chef is installed by crowbar_join
+<% if @target_platform_version.to_f < 12.0 %>
```
Comment (vuntz) [inline chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: Same here.

**Comment by MaximilianMeister** [comment]
@vuntz what is the difference between `%>` and `-%>`? i see `%>` in a few other places as well

File: chef/cookbooks/provisioner/templates/suse/crowbar_register.erb [hunk 3]
```diff
@@ -311,7 +316,11 @@ sed -i -e "s/\(127\.0\.0\.1.*\)/127.0.0.1 localhost.localdomain localhost/" /etc
 echo "$CROWBAR_KEY" > /etc/crowbar.install.key
 
 # normally done by crowbar_join, but only when chef is installed by crowbar_join
+<% if @target_platform_version.to_f < 12.0 -%>
 chkconfig -a chef-client
+<% else %>
```
Comment (MaximilianMeister) [inline chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: @vuntz here `-%>` too?

File: chef/cookbooks/provisioner/templates/suse/crowbar_register.erb [hunk 4]
```diff
@@ -311,7 +316,11 @@ sed -i -e "s/\(127\.0\.0\.1.*\)/127.0.0.1 localhost.localdomain localhost/" /etc
...
+systemctl enable chef-client.service
+<% end %>
```
Comment (MaximilianMeister) [inline chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: @vuntz here `-%>` too?

**Comment by MaximilianMeister** [comment]
i see its probably better to avoid line breaks with `-%>`

**Comment by dirkmueller** [comment]
LGTM

**Pull request merged**